crates/red_knot/src/types.rs

```diff
@@ -22,10 +22,12 @@ pub enum Type {
     Unknown,
     /// name is not bound to any value
     Unbound,
```

Comment:
This isn't really related to your PR, but I'm kinda leery of thinking of this as a "type", FWIW (I know pyright does this, but I don't like it much). Whether a symbol is bound or not in a given scope is obviously important for us to track, but it doesn't feel like it has much to do with the inferred _type_ of the symbol to me

Reply:
I keep going back and forth on this, I see arguments both ways. Like you say, it's not related to this PR, and nothing is using this yet anyway, so there's not an urgency to decide here, but we should discuss.